### PR TITLE
Performance metrics

### PR DESCRIPTION
**Pull request by Brechtpd**
99% of time of the witness generation is spent in RPC calls. At least when the node is remote. Hopefully when the node is local it is very fast, but still to be tested.

### CODE REVIEW

File: lib/src/lib.rs [hunk 1]
```diff
@@ -33,6 +33,48 @@ pub mod protocol_instance;
 pub mod prover;
 pub mod taiko_utils;
 
+#[cfg(not(target_os = "zkvm"))]
+mod time {
+    pub use core::ops::AddAssign;
+    pub use std::time::{Duration, Instant};
+}
+#[cfg(target_os = "zkvm")]
```
Comment (CeciliaZ030) [inline lib/src/lib.rs:41]: Is there issues if we use std::time in zkvm?

File: host/src/prover/execution.rs [hunk 2]
```diff
@@ -20,12 +20,18 @@ pub async fn execute<D: Prover>(
...
-    let elapsed = Instant::now().duration_since(start).as_millis() as i64;
-    observe_input(elapsed);
-    // 2. pre-build the block
+    let time_elapsed = Instant::now().duration_since(start);
+    println!(
+        "=> Witness generated in {}.{} seconds",
+        time_elapsed.as_secs(),
+        time_elapsed.subsec_millis()
+    );
+    observe_input(time_elapsed.as_millis() as i64);
```
Comment (CeciliaZ030) [inline host/src/prover/execution.rs:32]: shall we remove this? useless :(
and perhaps the whole https://github.com/taikoxyz/raiko/blob/d378dd491957b59f943091262836abc0fc709285/host/src/main.rs#L20

**Review by CeciliaZ030 — APPROVED** [review]
LGTM! 
I feel like we might need to refine it one more tim with something like https://crates.io/crates/tracing cuz now how we time things are very elementary and creates lots of extra code. 
I'll remove the metrics.rs when I clean the Cache so let's just merge this 👍

File: lib/src/lib.rs (hunk 1 above)
Comment (Brechtpd) [reply lib/src/lib.rs:41]: Yep! Halts in some time syscall.

File: host/src/prover/execution.rs (hunk 2 above)
Comment (Brechtpd) [reply host/src/prover/execution.rs:32]: Yeah kept it in for now because I am not sure of the purpose, removing it works for me.

**Comment by Brechtpd** [comment]
> I feel like we might need to refine it one more tim with something like https://crates.io/crates/tracing cuz now how we time things are very elementary and creates lots of extra code.

I'm not sure how that would save much code. There will be places where we want to start measuring stuff so we need some code there, and then some code at the place we want to stop measuring. Which the current code does quite minimally. So don't really see how you can significantly reduce the amount of code except some very small helper functions to combine stop + print time or something like that.

**Pull request merged**